cmd/promtool/unittest.go

```diff
@@ -0,0 +1,476 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/rules"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// RulesUnitTest does unit testing of rules based on the unit testing files provided.
+// An example file as input can be found here: https://gist.github.com/codesome/f4426d28244ee9fac128031ed2f9af70
```

Comment:
We should add proper docs for this

Reply:
Yes. I will add docs for this. So I will remove that comment from here as docs will be added for this.

Reply:
Oh, I was trying to add docs in https://github.com/prometheus/docs. Just found that there is a `docs/` directory in this repo. I will add them.

Reply:
I have added the docs.